### PR TITLE
node selector rework

### PR DESCRIPTION
**Pull request by Kostov6**
**What this PR does / why we need it**:
Reworks the git node selector

**Release note**:

```improvement developer
The git node selector's missing argument processing has been added
```

### CODE REVIEW

File: pkg/resourcehandlers/git/git_resource_handler.go [hunk 1]
```diff
@@ -150,53 +157,90 @@ func (g *Git) Accept(uri string) bool {
...
+			SHAAlias: rl.SHAAlias,
+			Type:     resourceType,
+			Path:     relativePath,
+		})
+		return nil
+	})
+	if err != nil {
+		return nil, err
 	}
+	/*
```
Comment (Kristian-ZH) [inline pkg/resourcehandlers/git/git_resource_handler.go:217]: It is not a good practice to leave a commented code.
If you do not need this, you can just remove it

File: pkg/resourcehandlers/github/resource_locator_cache.go [hunk 2]
```diff
@@ -193,6 +198,7 @@ func (c *Cache) set(ctx context.Context, rl *ResourceLocator) (bool, error) {
 		}
 	}
 	// add root repo key if not already added
+	//???
```
Comment (Kristian-ZH) [inline pkg/resourcehandlers/github/resource_locator_cache.go:201]: Why do you need this comment?

**Review by Kristian-ZH — CHANGES_REQUESTED** [review]
Some minor comments

**Comment by Kostov6** [comment]
>

File: pkg/resourcehandlers/git/git_resource_handler.go [hunk 3]
```diff
@@ -150,53 +157,64 @@ func (g *Git) Accept(uri string) bool {
...
-	for path, n := range nb.someMap {
-		parentPath := filepath.Dir(path)
-		parent, exists := nb.someMap[parentPath]
-		if !exists { // If a parent does not exist, this is the root.
-			_node = n
-		} else {
-			n.SetParent(parent)
-			n.Parent().Nodes = append(n.Parent().Nodes, n)
+	nodesSelectorLocalPath := tE.gitRH.getNodeSelectorLocalPath(repositoryPath, rl)
+	//var files []string
```
Comment (Kristian-ZH) [inline pkg/resourcehandlers/git/git_resource_handler.go:185]: Can you please remove these two commented lines

File: pkg/resourcehandlers/git/git_resource_handler.go [hunk 4]
```diff
@@ -150,53 +157,64 @@ func (g *Git) Accept(uri string) bool {
...
+	err := filepath.Walk(root, func(path string, info os.FileInfo, err error) error {
+		if path == root {
+			return nil
 		}
+		relativePath := strings.TrimPrefix(path, root+"/")
+		//dont include files from the .git folder, because they are not part of the repository
+		if relativePath == ".git" || strings.HasPrefix(relativePath, ".git/") {
+			return nil
+		}
+		//files = append(files, relativePath)
```
Comment (Kristian-ZH) [inline pkg/resourcehandlers/git/git_resource_handler.go:198]: same

**Review by dimitar-kostadinov — APPROVED** [review]
/lgtm

**Pull request merged**